shared/src/business/useCases/docketEntry/updateDocketEntryMetaInteractor.ts

```diff
@@ -191,20 +171,16 @@ export const updateDocketEntryMetaInteractor = async (
 
     caseEntity.updateDocketEntry(updatedDocketEntry);
   } else if (shouldRemoveExistingCoverSheet) {
-    await applicationContext.getPersistenceGateway().updateDocketEntry({
-      applicationContext,
-      docketEntryId: docketEntryEntity.docketEntryId,
-      docketNumber,
-      document: docketEntryEntity.validate(),
-    });
-    const updatedDocketEntry = await applicationContext
+    const { numberOfPages } = await applicationContext
       .getUseCaseHelpers()
       .removeCoversheet(applicationContext, {
         docketEntryId: originalDocketEntry.docketEntryId,
         docketNumber: caseEntity.docketNumber,
-        filingDateUpdated,
       });
-    caseEntity.updateDocketEntry(updatedDocketEntry);
+
+    docketEntryEntity.setNumberOfPages(numberOfPages);
```

Comment:
Is this the main bug fix?

Reply:
No, good question, there are a lot of changes here.

The main bug fix was switching from `pdfDoc.getPages().length` to `pdfDoc.getPageCount()`. I observed that when removing or adding pages to the pdf document, the getPages method was often off by one, I did some digging into the source code and issues for documentation on the issue and found a few old "resolved" reports of the same behavior but nothing concrete.

Reply:
Ah okay, good to know - thanks. I thought that was clean up.